articles/sql-or-nosql-when-to-choose-what/index.md

```diff
@@ -0,0 +1,123 @@
+How do you decide which database model to choose? Which is better? Which database fits your business/application model? These two databases are great, and they can leave you in a dilemma of not knowing which to choose. Why chose this, and not the other and it is okay to be in that position.
+
+Let’s check some of the factors that will help you decide what fits your needs.
+
+#### When to use NoSQL
+
+- Fits well when a business model has a rapid growth of data, and the database has no clear schema definitions, especially when you are dealing with unstructured information. Rapid data growth needs highly flexible data models to scale out the needs of that application.
+- If your business model has no defined schema or the data requirements and the schemas are constantly evolving.
+- When dealing with IoT technology, NoSQL will be a great fit as it offers real-time insights for sensor data.
+- Real-time transaction pieces of information such as customer behaviors, analytics information, financial markets, and trading activities would be great under NoSQL.  NoSQL will help you to check and watch real-time information transactions.
+- When dealing with millions and millions of rows and columns, NoSQL will be a significant consideration as it provides fast read and writes access to big data.
+
+#### When to use SQL
+
+- The biggest win of SQL is when the legacy of your system needs a relational structure. SQL will be a natural fit. SQL thrives in such circumstances that need high-quality and well-managed relational database systems that are easy to setup.
+- If your application requires multi-row transactions such as accounting and budget systems
+- Pre-defined data structure with multi-row transactions which have set schemas such as inventory system that have distinct entities. SQL will be the match as it provides perfect nominalizations and simple queries that comply with ACID
+    ACID represents:
+    - A (atomicity) a database process transaction is either wholly executed or not executed at all.
+    - C (consistency) only valid data is recorded to the database, and if a transaction was invalid, this database reverts to its former state. The inconsistent transaction is aborted, and an error is created into the error log.
+    - I (isolation) a transaction is securely and independently processed without interfering another transaction.
+    - D (durability) if a transaction process is completed successfully all its changes are stored permanently.
+
+- If your system values data integrity, then worry out. SQL database system is designed to change the data type to the way you want them to be, i.e. integer, varchar, text, Boolean, date value, etc.
+
+Let’s look at the 2020 [annual developer survey](https://insights.stackoverflow.com/survey) trends conducted by StackOverflow involving over 65,000 respondents from over 180 countries.
+
+- *[Most popular](https://insights.stackoverflow.com/survey/2020#technology-databases-all-respondents4) databases*
+
```

Comment:
You could add a table that concludes by summarizing the differences between SQL and NoSQL databases.

To add a table in Markdown, check out the syntax [here](https://www.markdownguide.org/extended-syntax/#tables).

Reply:
@kimkimani Please do add a summary of the things that were discussed in the form of a table, this would ensure that readers end up with a crisp summary in their mind.

Reply:
@cyberShaw A table seems to be very much repetitive, I consider adding a conclusion instead, will that be okay?

Reply:
Yep, that works too.